docs/guides/migrations.md

```diff
@@ -0,0 +1,242 @@
+---
+title: Migrations
+toc: true
+weight: 200
+indent: true
+---
+
+# Migrations
+
+Crossplane is built on CustomResourceDefinition (CRD) objects that expose an API
+with a certain schema. As the capabilities of the controller reconciling a CRD
+changes, so does its schema, which is what the users and their systems interact
+with. You will find that each schema is enumerated by a [version](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/).
+
+Kubernetes provides a way to make sure users of CRDs are able to migrate
+to the new versions without breaking their existing systems in a way that give
+them enough time to make the changes to adopt the new version. The versions in
+this context have certain implications about the API stability guarantees that
+the owner of the CRD promises. Crossplane community follows those guarantees.
+For more details about what those guarantees are, see [API Versioning](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+
+In this document, the migration steps for the most common cases Crossplane users
+encounter will be covered.
+
+## Managed Resources
+
+There are mainly three classes of migration scenarios:
+
+* Changes that don't require version increase.
+* Version increased with conversion support.
+* Version increased with no conversion support.
+
+### Change of Schema Without Version Increase
+
+This class of changes include cases where the developer changed the schema in a
+way that is fully compatible with the old schema. For example, addition of a new
+optional field falls under this class.
+
+You don't need to take any action for these cases.
+
+### Version Increased With Conversion Support
+
+In this case, a new schema is introduced as a new API version and the developer
+provided a way for Kubernetes API server to perform conversions between the old
+and the new version. Keep in mind that not all incompatible changes require
+developer to provide conversion support, see [API Versioning](https://kubernetes.io/docs/reference/using-api/#api-versioning) to see in what cases
+they need to provide it.
+
+In this case, your systems are expected to continue to work after the upgrade
+without any immediate changes. In most cases, developers would deprecate the old
+version and warn you about its removal schedule in the release notes. The steps
+described below need to be taken before that removal date.
+
+The important thing to keep in mind here is that Kubernetes API server is able to
+serve multiple versions equally well but the object stored in its persistence
+layer has a single shape. The point of the steps below is that we'll have the
```

Comment:
is this true? i expected that when a new version is released, that it will typically become the new stored version, so we're telling users to update their manifests to use the new version, and that new version is what will be the 1 stored version in the api-server too.

I have that wrong? 🤔

Reply:
This is what we do but we don't have to, i.e. you can choose an old or even non-served schema as the storage one. The only restriction is that you can only ever have a single storage version. You can see the toggle [here](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#customresourcedefinitionversion-v1-apiextensions-k8s-io).

Reply:
Thanks @muvaf - so when we bump something from say v1alpha1 -> v1beta1, we leave `storage: false` on the new v1beta1 version?  When does the new v1beta1 version eventually becoming the `storage: true` version?

Maybe a little table example could be helpful in this doc, that very clearly shows the matrix of expected served/storage for each version involved in the migration? 😊